pylint/lint/pylinter.py

```diff
@@ -709,7 +710,15 @@ def _get_asts(
         """Get the AST for all given FileItems."""
         ast_per_fileitem: dict[FileItem, nodes.Module | None] = {}
 
+        # For progress reports, it would be nice to say "item 1 of N",
```

Comment:
Right, maybe we could try to call list on the iterators with a time out and either have items '1 of N', if it succeed or '1 of a lot' if we timeout ? Not very KISS.

Reply:
That's possible, but perhaps we could leave that for a future iteration.  I don't know what the time proportion of getting the ASTs vs doing the linting is.  If getting the ASTs is super slow, then a "1 of N" counter would be nice, but if linting is the performance killer, then not having the AST counter would be ok.

Calling `list` on the iterators would mean they'd have to be re-instantiated, and I didn't want to touch the existing code too much, it felt too invasive.

Reply:
ast inferring is the bottleneck for sure